File: core-banking/backend/module/business-logic/src/main/java/com/revy/example/order/command/dto/PlaceOrderCommand.java
package com.revy.example.order.command.dto;

import com.revy.example.domain.account.enums.OrderSide;
import com.revy.example.domain.account.enums.OrderType;

import java.math.BigDecimal;
import java.time.Instant;

public record PlaceOrderCommand(
        Long       accountId,
        Long       stockId,
        OrderSide  side,
        OrderType  orderType,
        BigDecimal quantity,
        BigDecimal limitPrice,   // null for MARKET
        String     referenceId,
        Instant    orderedAt
) {}

File: practice_kotlin/kotlin-trading-core/src/main/kotlin/com/revy/trading/application/port/input/payload/PlaceOrderCommand.kt
package com.revy.trading.application.port.input.payload

import com.revy.trading.domain.enums.OrderSide
import java.math.BigDecimal
import java.util.UUID

data class PlaceOrderCommand(
    val accountId: UUID,
    val symbol: String,
    val side: OrderSide,
    val quantity: Long,
    val limitPrice: BigDecimal,
)
